lib/rules/id-length.js

```diff
@@ -28,7 +30,7 @@ module.exports = function(context) {
                 // regular property assignment
                 parent.parent.left === parent || (
                     // or the last identifier in an ObjectPattern destructuring
-                    parent.parent.type === "Property" && parent.parent.value === parent &&
+                    parent.parent.type === "Property" && properties && parent.parent.value === parent &&
```

Comment:
>If you say bumping that check up doesn't create false positives/negatives then I can do that!

I don't, I rely on tests to catch that if it happens :) This part was hard for me to figure out and `git blame` shows I added it only to support ES6 destructuring stuff so my hunch is "it should be okay".